File: adspro_test/Models/Address.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Web;

namespace adspro_test.Models
{
    [Table("Addresses")]
    public class Address
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }

        [Required]
        public string Type { get; set; }

        [Required]
        public Guid StreetId { get; set; }
        public Street Street { get; set; }

        [Required]
        public string PostalCode { get; set; }

        [Required]
        public string City { get; set; }
    }
}

File: adspro_test/src/app/interfaces/address.ts
import { Street } from "./street";

export class Address {
    id: string;
    type: string;
    postalCode: string;
    city: string;
    street: Street;

    constructor() {
        this.id = '';
        this.type = '';
        this.postalCode = '';
        this.city = '';
        this.street = {
            id: '',
            street1: '',
            street2: ''
        }
    }
}
